tests/integration/main.go

```diff
@@ -131,9 +134,14 @@ func (tr *TestRun) runStep(step Step, verbose bool) {
 	}
 }
 
-// Starts docker container and sequentially runs steps
-func (tr *TestRun) ExecuteSteps(wg *sync.WaitGroup, steps []Step) {
+// ExecuteStepsInWaitGroup runs steps in a WaitGroup so that the test run can be run in parallel.
+func (tr *TestRun) ExecuteStepsInWaitGroup(wg *sync.WaitGroup, steps []Step) {
```

Comment:
Seems less bug prone if you don't require the caller to increment the wg themselves
```suggestion
func (tr *TestRun) ExecuteStepsInWaitGroup(wg *sync.WaitGroup, steps []Step) {
wg.Add(1)
```

Reply:
`ExecuteStepsInWaitGroup` was removed.

Both `wg.Add(1)` and `wg.Done()` are called from the top level after refactoring.
```golang
wg.Add(1)
go func(run testRunWithSteps) {
	defer wg.Done()
	tr := run.testRun
	tr.SetLocalSDKPath(*localSdkPath)
	tr.ValidateStringLiterals()
	tr.startDocker()
	tr.ExecuteSteps(run.steps)
	tr.teardownDocker()
}(run)
```